README.md

```diff
@@ -0,0 +1,88 @@
+# Clean microservice template
+This is a project template to create microservices in .NET 6 based on the Clean architecture approach (or Onion, or Hexagonal, or Ports and adapters).  
+
+
+## Main layers
+
+### API
+The API is generated using the new minimal api approach as the microservice shouldn't be that big
+
+### Application
+The application layer contains all queries and commands so you can see the operations this microservice is doing in a quick sight.  
+
+### Domain
+The domain layer uses records to represent our aggragte roots and value objects.
+
+### Infrastructure
+Only the database layer is implemented using an in memory repository.
+
+
+## TODO
+
+#### ADR decisions should be reachable from the solution explorer... how?
+#### Shall we add Id primitives for entities?
+https://andrewlock.net/using-strongly-typed-entity-ids-to-avoid-primitive-obsession-part-1/
+#### We give up with mapster for Value objects. Moved to a custome mapper. See Method GetAll from repository.
+
+#### What about the metrics? 
+AppMetrics doesn't seem to be updated for .NET 6.    
+There is this option from MS [Announcing dotnet monitor in .NET 6](https://devblogs.microsoft.com/dotnet/announcing-dotnet-monitor-in-net-6/)   
+
```

Comment:
More to-do's.....

- Improve validation
- Fix namespaces by adding company name and product/template name
- Add unit testing
- Add logging. 
     Scoped logging, propose the right places on where and what to log (Request filters, mediator behaviors, etc)
- Behavior pipelines for exception handling
- Use records
- Add more examples for Value objects
- Ubiquitous language. Generate dictionary from comments on the Domain layer
- More explanations on the readme or other docs
- ....

Reply:
I could work on making entities records and struct records ;) @robertsynoradzki be aware that could be some conflicts with the validation. Anyway, I would let you now before starting, maybe on Sunday 24th or probably on Monday 25th